tools/aiecc/aiecc/main.py

```diff
@@ -320,10 +324,12 @@ async def run_flow(self):
         progress.update(progress.task,advance=0,visible=False)
         progress.task_completed = progress.add_task("[green] AIE Compilation:", total=len(cores)+1, command="%d Workers" % nworkers)
 
-        processes = [self.process_arm_cgen()]
+        self.progress_bar = progress
+        processes = []
         for core in cores:
           processes.append(self.process_core(core))
         await asyncio.gather(*processes)
+        await self.process_arm_cgen()
 
```

Comment:
Why this change?  The intention here is to start running ARM code generation ASAP, since it is slower than everything else.  If we can't do that here because the generated ELF files need to be parsed, then we should probably generate a header and separately compile and link the host code in 2 parts.

Reply:
The change here is that the airbin generation occurs in `self.process_arm_cgen()` and it needs to have the AIE elfs already in order to generate the airbin

Reply:
I'd prefer to see this separated out.